server/src/main/java/org/elasticsearch/rest/DeprecationRestHandler.java

```diff
@@ -65,7 +65,7 @@ public DeprecationRestHandler(
     @Override
     public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
         if (compatibleVersionWarning == false) {
-            deprecationLogger.critical(DeprecationCategory.API, deprecationKey, deprecationMessage);
+            deprecationLogger.warn(DeprecationCategory.API, deprecationKey, deprecationMessage);
```

Comment:
We will still need a way to emit critical level deprecations when decalaring a REST handler. I will log a follow up PR.